src/rust/cryptography-x509/src/extensions.rs

```diff
@@ -285,9 +285,16 @@ impl KeyUsage<'_> {
     }
 }
 
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+pub struct NamingAuthority<'a> {
+    pub id: Option<asn1::ObjectIdentifier>,
+    pub url: Option<asn1::IA5String<'a>>,
+    pub text: Option<asn1::Utf8String<'a>>,
```

Comment:
I see this field being a `DirectoryString` in the spec. Why UTF8 only?

Reply:
Because I didn't pay enough attention, as usual :see_no_evil:  Thank you for the review - this is extremely helpful! Won't repeat the same mistakes in the subsequent PRs.